ops/ccs-ops-misc/synthetic-data/scripts/prepare-for-new-synthea-load.py

```diff
@@ -0,0 +1,200 @@
+        print(f"Carrier claims group id potential conflict, {result} values past start value {clm_grp_id_start} across all tables")
+        ranges_good = False
+    else:
+        print("Carrier claims group id range is valid")
+        
+    pde_id_start = re.findall("exporter.bfd.pde_id_start.*$",properties_file,re.MULTILINE)[0].split("=")[1]
+    query = f"select count(*) from partd_events where pde_id::bigint <= {pde_id_start}"
+    result = _execute_single_count_query(db_string, query)
+    if result > 0:
+        print(f"PDE id potential conflict, {result} values after start {pde_id_start}")
+        ranges_good = False
+    else:
+        print("PDE id range is valid")
+    
+    carr_clm_ctrl_num_start = re.findall("exporter.bfd.carr_clm_cntl_num_start.*$",properties_file,re.MULTILINE)[0].split("=")[1]
+    query = f"select count(*) from carrier_claims where carr_clm_cntl_num::bigint <= {carr_clm_ctrl_num_start}"
+    result = _execute_single_count_query(db_string, query)
+    if result > 0:
+        print(f"carr_clm_cntl_num id range invalid, {result} values past starting id {carr_clm_ctrl_num_start}")
+        ranges_good = False
+    else:
+        print("carr_clm_cntl_num range is valid")
+    
+    ## Check fi_num_start in all the tables it exists in
+    fi_num_start = re.findall("exporter.bfd.fi_doc_cntl_num_start.*$",properties_file,re.MULTILINE)[0].split("=")[1]
+    query = f"select count(*) from outpatient_claims where fi_doc_clm_cntl_num::bigint <= {fi_num_start}"
+    result = _execute_single_count_query(db_string, query)
+    
+    fi_num_start = re.findall("exporter.bfd.fi_doc_cntl_num_start.*$",properties_file,re.MULTILINE)[0].split("=")[1]
+    query = f"select count(*) from inpatient_claims where fi_doc_clm_cntl_num::bigint <= {fi_num_start}"
+    result = result + _execute_single_count_query(db_string, query)
+    
+    fi_num_start = re.findall("exporter.bfd.fi_doc_cntl_num_start.*$",properties_file,re.MULTILINE)[0].split("=")[1]
+    query = f"select count(*) from hha_claims where fi_doc_clm_cntl_num::bigint <= {fi_num_start}"
+    result = result + _execute_single_count_query(db_string, query)
+    
+    fi_num_start = re.findall("exporter.bfd.fi_doc_cntl_num_start.*$",properties_file,re.MULTILINE)[0].split("=")[1]
+    query = f"select count(*) from snf_claims where fi_doc_clm_cntl_num::bigint <= {fi_num_start}"
+    result = result + _execute_single_count_query(db_string, query)
+    
+    fi_num_start = re.findall("exporter.bfd.fi_doc_cntl_num_start.*$",properties_file,re.MULTILINE)[0].split("=")[1]
+    query = f"select count(*) from hospice_claims where fi_doc_clm_cntl_num::bigint <= {fi_num_start}"
+    result = result + _execute_single_count_query(db_string, query)
+    
+    if result > 0:
+        print(f"fi_doc_cntl_num id range invalid, {result} values past starting id {fi_num_start} across all tables")
+        ranges_good = False
+    else:
+        print("fi_doc_cntl_num range is valid")
```

Comment:
Why not have this check and explicit message for each of the claim types like what you have for carrier, dme, and benes?

Reply:
possibly even a function to abstract this

Reply:
Good call; I've updated the output to be more precise, abstracted the logic, and added it for all the tables/fields checked individually.